### PR TITLE
Components: Prevent broken lists in auto-generated readmes

### PR DESCRIPTION
**Pull request by mirka**
## What?

Prevents broken lists (`ul`) in the auto-generated component READMEs, by inverting the order of the "prop data" `ul` and the prop description.

## Why?

Because we can assume that a prop description will always start with a paragraph (not `ul`), flipping the order here should solve the "ambiguous run-on list" problem when the prop description ends with a list.

### Before

```md
### `myProp`

This is my prop description. The prop does three things:

- Foo
- Bar
- Baz

- Type: `string`
- Required: No
```

The ambiguously separated Markdown lists will render unreliably, depending on the HTML converter implementation.

### After

```md
### `myProp`

- Type: `string`
- Required: No

This is my prop description. The prop does three things:

- Foo
- Bar
- Baz
```

Now the lists are clearly separated.

## Testing Instructions

The READMEs re-generated by `npm run docs:components` do not include ambiguous lists, even when a prop description ends with a list.

### CODE REVIEW

File: bin/api-docs/gen-components-docs/markdown/index.mjs [hunk 1]
```diff
@@ -9,17 +9,14 @@ import json2md from 'json2md';
...
  */
-function normalizeTrailingNewline( str ) {
-	if ( ! str?.trim() ) {
-		return undefined;
-	}
-	return str.replace( /\n*$/, '\n' );
-}
+json2md.converters.md = ( input ) => {
+	return input?.trim() || '';
+};
```
Comment (mirka) [inline bin/api-docs/gen-components-docs/markdown/index.mjs:19]: I realized you can add custom converters like this, which makes things more simpler and readable overall. When you use a converter, `json2md` adds a trailing newline automatically.

The removal of redundant newlines is moved to the end of this file, which will clean the entire output.

Comment (tyxla) [reply bin/api-docs/gen-components-docs/markdown/index.mjs:19]: TIL, this is very cool!

File: bin/api-docs/gen-components-docs/markdown/index.mjs [hunk 2]
```diff
@@ -49,5 +46,5 @@ export function generateMarkdownDocs( { typeDocs, subcomponentTypeDocs } ) {
 
 	return json2md(
 		[ ...mainDocsJson, ...subcomponentDocsJson ].filter( Boolean )
-	);
+	).replace( /\n+$/gm, '\n' ); // clean unnecessary consecutive newlines
```
Comment (tyxla) [inline bin/api-docs/gen-components-docs/markdown/index.mjs:49]: Noting that this cleans up only trailing ones, maybe we want to update the comment to reflect that?

**Review by tyxla — APPROVED** [review]
LGTM, nice 👍

File: bin/api-docs/gen-components-docs/markdown/index.mjs (hunk 2 above)
Comment (mirka) [reply bin/api-docs/gen-components-docs/markdown/index.mjs:49]: [This](https://regex101.com/r/YHxZ1L/2) is a `global` + `multiline` regex, intended to remove all excessive newlines within in the entire readme output.

**Pull request merged**